<a href="https://colab.research.google.com/github/Zollicoff/physics_fall_2022/blob/main/Engineering_Physics_I_Experiment_1_Addition_of_Vectors_Experiment.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Engineering Physics I - Experiment 1 - Addition of Vectors Experiment**

Report written by Zachary A. Hampton

List of group members: Adolpho Magallanes, Brandon Swift, Ivan Cuevas, Oscar Lara Garcia

Experiment performed on 09/07/2022

No Current Due Date

# Setup

First let us import Numpy for mathematical calculations and set options to suppress scientific notation.

And let us define the gravity constant of 9.8 m/s as G.

In [1]:
import numpy as np
np.set_printoptions(suppress=True)
G = 9.8

# Part 1 - Two Applied Forces

Let us define the two given vectors as Vector Array 1 and Vector Array 2

Let us also define our Measurements of the Counter-Weight as Vector Array FE1:

(Where[0] = Mass in kg, [1] = Gravity in m/s, [2] = Angle in degrees)

In [2]:
vectorarray1 = np.array([0.10, G, 20])
vectorarray2 = np.array([0.20, G, 90])
vectorarrayfe1 = np.array([.25, G, 249])
print('Vector Array 1: Mass = ', vectorarray1[0], 'kg, Gravity = ', vectorarray1[1], 'm/s, Angle = ', vectorarray1[2], 'degrees')
print('Vector Array 2: Mass = ', vectorarray2[0], 'kg, Gravity = ', vectorarray2[1], 'm/s, Angle = ', vectorarray2[2], 'degrees')
print('Vector Array FE1: Mass = ', vectorarrayfe1[0], 'kg, Gravity = ', vectorarrayfe1[1], 'm/s, Angle = ', vectorarrayfe1[2], 'degrees')

Vector Array 1: Mass =  0.1 kg, Gravity =  9.8 m/s, Angle =  20.0 degrees
Vector Array 2: Mass =  0.2 kg, Gravity =  9.8 m/s, Angle =  90.0 degrees
Vector Array FE1: Mass =  0.25 kg, Gravity =  9.8 m/s, Angle =  249.0 degrees


Let us define FR1 as the opposite angle of FE1 by subtracting 180 from FE1:

In [3]:
fr1 = int(vectorarrayfe1[2]) - 180
print('The result is ' + str(fr1) + ' degrees.')

The result is 69 degrees.


Calculate force of Vector Array 1 as f1 and Vector Array 2 as f2, as well as Vector Array FE1 / FR1 *in* Newtons by multiplying Mass and Gravity:

In [4]:
f1 = vectorarray1[0] * vectorarray1[1]
f2 = vectorarray2[0] * vectorarray2[1]
fe1 = vectorarrayfe1[0] * vectorarrayfe1[1]
print('Force of Vector Array 1 =', f1, 'Newtons')
print('Force of Vector Array 2 =', f2, 'Newtons')
print('Force of Vector Array FE1 / FR1 =', fe1, 'Newtons')

Force of Vector Array 1 = 0.9800000000000001 Newtons
Force of Vector Array 2 = 1.9600000000000002 Newtons
Force of Vector Array FE1 / FR1 = 2.45 Newtons


Calculate FX1 / FX2 using cos and FY1 / FY2 using sin:

In [5]:
fx1 = f1 * np.cos(np.radians(vectorarray1[2]))
fy1 = f1 * np.sin(np.radians(vectorarray1[2]))
fx2 = f2 * np.allclose(0, np.cos((vectorarray2[2])), 1e-5, 1e-8)
fy2 = f2 * np.sin(np.radians(vectorarray2[2]))
print('FX1 =', fx1)
print('FY1 =', fy1)
print('FX2 =', fx2)
print('FY2 =', fy2)

FX1 = 0.9208987683701904
FY1 = 0.3351797404591554
FX2 = 0.0
FY2 = 1.9600000000000002


Define A by adding FX1 and FX2, and B by adding FY1 and FY2 to use in Pythagorean Theorem:

In [6]:
a1 = fx1 + fx2
b1 = fy1 + fy2
print('A =', a1)
print('B =', b1)

A = 0.9208987683701904
B = 2.2951797404591554


Use Pythagorean Theorem pythan $a^2 + b^2 = c^2$ to find C:



In [7]:
c1 = np.hypot(a1, b1)
print('C^2 = ' + str(a1) + '^2 ' + str(b1) + '^2')
print('C = ', c1)

C^2 = 0.9208987683701904^2 2.2951797404591554^2
C =  2.473035499664307


Find the Error Rate Percentage with $100(FE1-C)/C$:

In [8]:
error1 = (fe1 - c1) / c1
errorpercent1 = error1 * 100
print(error1)
print('The Error Rate is', errorpercent1, '%')


-0.009314665991423828
The Error Rate is -0.9314665991423828 %


# Part 2 - Three Applied Forces

Let us define the Three given vectors as Vector Array 3, Vector Array 4, and Vector Array 5.

Let us also define our Measurements of the Counter-Weight as Vector Array FE2:

(Where[0] = Mass in kg, [1] = Gravity in m/s, [2] = Angle in degrees)

In [9]:
vectorarray3 = np.array([0.15, G, 30])
vectorarray4 = np.array([0.20, G, 100])
vectorarray5 = np.array([0.10, G, 145])
vectorarrayfe2 = np.array([0.32, G, 268])
print('Vector Array 3: Mass = ', vectorarray3[0], 'kg, Gravity = ', vectorarray3[1], 'm/s, Angle = ', vectorarray3[2], 'degrees')
print('Vector Array 4: Mass = ', vectorarray4[0], 'kg, Gravity = ', vectorarray4[1], 'm/s, Angle = ', vectorarray4[2], 'degrees')
print('Vector Array 5: Mass = ', vectorarray5[0], 'kg, Gravity = ', vectorarray5[1], 'm/s, Angle = ', vectorarray5[2], 'degrees')
print('Vector Array FE2: Mass = ', vectorarrayfe2[0], 'kg, Gravity = ', vectorarrayfe2[1], 'm/s, Angle = ', vectorarrayfe2[2], 'degrees')

Vector Array 3: Mass =  0.15 kg, Gravity =  9.8 m/s, Angle =  30.0 degrees
Vector Array 4: Mass =  0.2 kg, Gravity =  9.8 m/s, Angle =  100.0 degrees
Vector Array 5: Mass =  0.1 kg, Gravity =  9.8 m/s, Angle =  145.0 degrees
Vector Array FE2: Mass =  0.32 kg, Gravity =  9.8 m/s, Angle =  268.0 degrees


Let us define FR2 as the opposite angle of FE2 by subtracting 180 from FE2:

In [10]:
fr2 = int(vectorarrayfe2[2]) - 180
print('The result is ' + str(fr2) + ' degrees.')

The result is 88 degrees.


Calculate force of Vector Array 3 as f3, Vector Array 4 as f4, Vector Array 5 as f5, as well as Vector Array FE2 / FR2 in Newtons by multiplying Mass and Gravity:

In [11]:
f3 = vectorarray3[0] * vectorarray3[1]
f4 = vectorarray4[0] * vectorarray4[1]
f5 = vectorarray5[0] * vectorarray5[1]
fe2 = vectorarrayfe2[0] * vectorarrayfe2[1]
print('Force of Vector Array 3 =', f3, 'Newtons')
print('Force of Vector Array 4 =', f4, 'Newtons')
print('Force of Vector Array 5 =', f5, 'Newtons')
print('Force of Vector Array FE2 / FR2 =', fe2, 'Newtons')

Force of Vector Array 3 = 1.47 Newtons
Force of Vector Array 4 = 1.9600000000000002 Newtons
Force of Vector Array 5 = 0.9800000000000001 Newtons
Force of Vector Array FE2 / FR2 = 3.136 Newtons


Calculate FX3 / FX4 / FX5 using cos and FY3 / FY4 / FY5 using sin:

In [12]:
fx3 = f3 * np.cos(np.radians(vectorarray3[2]))
fy3 = f3 * np.sin(np.radians(vectorarray3[2]))
fx4 = f4 * np.cos(np.radians(vectorarray4[2]))
fy4 = f4 * np.sin(np.radians(vectorarray4[2]))
fx5 = f5 * np.cos(np.radians(vectorarray5[2]))
fy5 = f5 * np.sin(np.radians(vectorarray5[2]))
print('FX3 =', fx3)
print('FY3 =', fy3)
print('FX4 =', fx4)
print('FY4 =', fy4)
print('FX5 =', fx5)
print('FY5 =', fy5)

FX3 = 1.273057343563125
FY3 = 0.7349999999999999
FX4 = -0.3403504282271834
FY4 = 1.930223195903928
FX5 = -0.8027690034032121
FY5 = 0.5621049076240251


Define A by adding FX3, FX4, and FX5, and B by adding FY3, FY4, and FY5 to use in Pythagorean Theorem:

In [13]:
a2 = fx3 + fx4 + fx5 
b2 = fy4 + fy4 + fy5
print('A =', a2)
print('B =', b2)

A = 0.1299379119327294
B = 4.422551299431881


Use Pythagorean Theorem $a^2 + b^2 = c^2$ to find C:

In [14]:
c2 = np.hypot(a2, b2)
print('C^2 = ' + str(a2) + '^2 ' + str(b2) + '^2')
print('C = ', c2)

C^2 = 0.1299379119327294^2 4.422551299431881^2
C =  4.424459724877611


Find the Error Rate Percentage with $100(FE1-C)/C$:

In [15]:
error2 = (fe2 - c2) / c2
errorpercent2 = error2 * 100
print(error2)
print('The Error Rate is', errorpercent2, '%')


-0.29121289490622543
The Error Rate is -29.121289490622544 %


# Conclusion and Questions:

Comparing the experimental measurements to the actual math, I was able to calculate the error rate for each version of the experiment.

On the first experiment with two applied forces I had an error rate of .931%.

And on the second experiment with three applied forces I had an error rate of 29.121%.

This error rate is a slight cause for concern since it is so high. I feel like my group did have equilibrium on the center ring, so maybe the wiggle room for balance on three is higher than previously expected.